podcasts/StreamingEpisodeArtwork.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+import Kingfisher
+
+/// Extracts artwork from a streaming episode (if there's any)
+class StreamingEpisodeArtwork {
+    static let shared = StreamingEpisodeArtwork()
+
+    private lazy var subscribedPodcastsCache: ImageCache = {
+        let path = (NSHomeDirectory() as NSString).appendingPathComponent("Documents/artworkv3")
+        let url = URL(fileURLWithPath: path)
+        subscribedPodcastsCache = try! ImageCache(name: "subscribedPodcastsCache", cacheDirectoryURL: url)
+        subscribedPodcastsCache.diskStorage.config.sizeLimit = UInt(400.megabytes)
+        subscribedPodcastsCache.diskStorage.config.expiration = .days(365)
+        return subscribedPodcastsCache
+    }()
+
+    /// Extract a UIImage from a given asset if embedded artwork option is enabled.
+    /// If an image is extracted, `episodeEmbeddedArtworkLoaded` notification is triggered
+    /// - Parameters:
+    ///   - asset: an AVAsset
+    ///   - episodeUuid: the UUID of the current playing episode
+    func loadEmbeddedImage(asset: AVAsset, episodeUuid: String) {
+        guard Settings.loadEmbeddedImages, !isCached(episodeUuid: episodeUuid) else {
+            return
+        }
+
+        let metadata = asset.metadata
+        for item in metadata {
+            if let key = item.commonKey?.rawValue, key == "artwork" {
+                if let imageData = item.value as? Data {
+                    let image = UIImage(data: imageData)
+
+                    if let image {
+                        set(for: episodeUuid, image: image) {
+                            NotificationCenter.postOnMainThread(notification: Constants.Notifications.episodeEmbeddedArtworkLoaded)
+                        }
+                    }
+
+                }
+            }
+        }
+    }
```

Comment:
Looking at the [AVFoundation docs](https://developer.apple.com/documentation/avfoundation/media_assets/retrieving_media_metadata) they recommend using [AVMetadataItem.metadataItems(from:filteredByIdentifier:)](https://developer.apple.com/documentation/avfoundation/avmetadataitem/1385843-metadataitems), which means we can reduce this down to:

```suggestion
        let artworkItems = AVMetadataItem.metadataItems(from: asset.commonMetadata, filteredByIdentifier: .commonIdentifierArtwork)
        let image = artworkItems.compactMap { $0.dataValue.flatMap { UIImage(data: $0) } }.first

        guard let image else { return }

        set(for: episodeUuid, image: image) {
            NotificationCenter.postOnMainThread(notification: Constants.Notifications.episodeEmbeddedArtworkLoaded)
        }
```

Reply:
Done in 8e0ee5b63e2f69351c50fcc9dc4ea973c02f0ddf